client/quantum_serverless/core/job.py

```diff
@@ -835,3 +803,101 @@ def _map_status_to_serverless(status: str) -> str:
         return status_map[status]
     except KeyError:
         return status
+
+
+def _upload_with_docker_image(
+    program: QiskitFunction, url: str, token: str, span: Any
+) -> str:
+    """Uploads function with custom docker image.
+
+    Args:
+        program (QiskitFunction): function instance
+        url (str): upload gateway url
+        token (str): auth token
+        span (Any): tracing span
+
+    Returns:
+        str: uploaded function name
+    """
+    response_data = safe_json_request(
+        request=lambda: requests.post(
+            url=url,
+            data={
+                "title": program.title,
+                "image": program.image,
+                "arguments": json.dumps({}),
```

Comment:
unrelated question.  Is the `arguments` for the `program` used?

Reply:
probably you are right and it is not used anymore. I will look it up and try to fix it in following PRs